optaplanner-core/src/main/java/org/optaplanner/core/impl/exhaustivesearch/node/comparator/ScoreFirstNodeComparator.java

```diff
@@ -35,8 +35,8 @@ public ScoreFirstNodeComparator(boolean scoreBounderEnabled) {
 
     @Override
     public int compare(ExhaustiveSearchNode a, ExhaustiveSearchNode b) {
-        // Investigate better score first
-        int scoreComparison = a.getScore().compareTo(b.getScore());
+        // Investigate better score first (ignore initScore to avoid depth first ordering)
```

Comment:
Done (it was a small fix, so I've done it quickly meself), see http://github.com/droolsjbpm/optaplanner/commit/215351423